core/tx_pool.go

```diff
@@ -1824,6 +1860,59 @@ func (t *txLookup) RemotesBelowTip(threshold *big.Int) types.Transactions {
 	return found
 }
 
+type timestampedTxHashSet struct {
+	lock       sync.RWMutex
+	hashes     []common.Hash
+	timestamps map[common.Hash]time.Time
+	ttl        time.Duration
+}
+
+func newExpiringTxHashSet(ttl time.Duration) *timestampedTxHashSet {
+	s := &timestampedTxHashSet{
+		hashes:     make([]common.Hash, 0),
+		timestamps: make(map[common.Hash]time.Time),
+		ttl:        ttl,
+	}
+
+	return s
+}
+
+func (s *timestampedTxHashSet) Add(hash common.Hash) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	s.hashes = append(s.hashes, hash)
+	s.timestamps[hash] = time.Now().Add(s.ttl)
```

Comment:
Looks like anything added as a private transaction will _always_ make the hash linger for three days, even if the tx was mined two days ago. Maybe consider pruning this on new blocks too?

Reply:
Yeah, we'll def do this when we work on another version of this –– I think this isn't too big of a memory concern right now, since the private transaction volume isn't huge and only hashes are being stored here. Main consideration previously was keeping this implementation sample simple.

Reply:
I went back to fix up this suggestion today and I remembered another reason I didn't want to implement pruning on blocks. I was trying to keep the CPU time of this struct extremely minimal by only ever pruning straight along the `s.hashes` until the last expired entry to avoid iterating over all the hashes on each. In addition, private transactions tend to take longer to get confirmed compared to normal transactions (limited hash power) so the benefit of cleaning the private transaction hashes is a bit less vs. the normal mempool.